datalad/core/local/save.py

```diff
@@ -206,6 +206,8 @@ def __call__(path=None, message=None, dataset=None,
                 recursive=recursive,
                 recursion_limit=recursion_limit,
                 on_failure='ignore',
+                # for save without recursion only commit matters
+                eval_subdataset_state='full' if recursive else 'commit',
```

Comment:
@mih proposing the same change in gh-4531 is a pretty good approval of this change :)

Reply:
Indeed ;-) but may be he would come up even work s better one which would work for recursive with a limit?

Reply:
Because you bring up recursion limits: Have you ever seen any actual use of it? I mean outside of tests. My shell history doesn't have it. This feels like a historical artifact of "why not have a limit too". It causes quite a few complications by requiring a relatively high level of sophistication in some code pieces. I am afraid that we will add another one here.

Reply:
consciously -- I do not recollect using it.  My eternal shell history setup is broken, but on smaug I found entire 2 invocations for `diff`:
```
datalad diff -r --recursion-limit 1 openneuro
datalad diff -r --recursion-limit 2 openneuro
```
which kinda makes sense BUT could have been worked around by `-C` or `-d` `openneuro` and not using `-r`. So indeed -- I don't have immediate use cases.  Moreover I think it would make sense only in "homogeneous" hierarchy, e.g. like HCP dataset (if it had more subdataset levels ;-)) -- probably a rare use case on its own. In all others it would likely be on a single path/subdataset, thus could be workedaround.

Reply:
gy, search on github brings up https://github.com/recoveringyank/T1ify/blob/1772167b3d771632441377fc47c729cc33d82a4b/get_data_datalad.sh with
```
datalad install ///hcp-openaccess -r --recursion-limit 3
```
when I [exclude some repos](https://github.com/search?q=datalad+recursion+limit+-repo%3Adartmouth-pbs%2Fdatalad-private+-repo%3Adatalad%2Fdatalad-private+-repo%3Adatalad%2Fgit-annex-ria-remote+-repo%3Acmdsearchcom%2Fcmdsearchcom.github.io&type=Code) I still do not see more of 3rd party relevant hints arriving... so at least nobody uses it heavily in the code besides us interfacing from one function to another ;)